geopandas/base.py

```diff
@@ -288,9 +291,8 @@ def total_bounds(self):
 
     @property
     def sindex(self):
-        if not self._sindex_valid:
+        if not self._sindex:
             self._generate_sindex()
-            self._sindex_valid = True
```

Comment:
I think the reason that `_sindex_valid` is used here instead of `self._sindex`, is to prevent that you try to rebuild the sindex each time when that fails (because in that case, `self._sindex` stays None).